src/renderers/dom/server/__tests__/ReactServerRendering-test.js

```diff
@@ -400,4 +400,26 @@ describe('ReactServerRendering', function() {
       expect(markup.indexOf('hello, world') >= 0).toBe(true);
     });
   });
+
+  it('warns with a no-op when an async setState is triggered', function() {
+    var Component = React.createClass({
+      componentWillMount: function() {
+        setTimeout(() => {
+          this.setState({});
+        });
+      },
+      render: function() {
+        return <div onClick={() => {}} />;
+      },
+    });
+
+    ReactReconcileTransaction.prototype.perform = function() {
```

Comment:
Tests should be written against the public API. You shouldn't need to override anything just to test it. Test public behavior instead—such as that calling `setState`, on the contrary, does not throw (or what was the original fiddle doing?)